File: scala/006/01.scala

def sumOfSquares(xs: Seq[Int]): Long = {
  xs.foldLeft(0L)((sum, x) => sum + Math.round(Math.pow(x, 2)))
}

def squareOfSum(xs: Seq[Int]): Long = {
  var sum = xs.foldLeft(0L)((sum, x) => sum + x)
  Math.round(Math.pow(sum, 2))
}

val range = 1 to 100
println(squareOfSum(range) - sumOfSquares(range))

File: php/006/01.php
<?
    $n = $argv[1];

    if ($n == '') {
        $n = 100;
    }

    $sum_of_squares = 0;
    $square_of_sum  = 0;

    for ( $i = 1; $i <= $n; $i++ ) {
        $sum_of_squares += $i*$i;
    }

    $square_of_sum = pow(($n * ($n+1) / 2),2);

    $diff = abs($sum_of_squares - $square_of_sum);

    printf("%d\n", $diff);
?>
